lib/chef/resource/windows_certificate.rb

```diff
@@ -299,6 +301,11 @@ def format_raw_out(out)
           end_cert = "-----END CERTIFICATE-----"
           begin_cert + out[/#{begin_cert}(.*?)#{end_cert}/m, 1] + end_cert
         end
+
+        # Checks if the certificate is binary encoded or not
+        def binary_cert?
+          powershell_out("file -b --mime-encoding #{new_resource.source}").stdout.strip == "binary"
```

Comment:
I'm fairly certain we can do this in pure ruby as well and avoid the extra time it takes to spin up a powershell instance

Reply:
Right @tas50 , one way of doing this could be by checking `String.encoding` which is not that reliable or we can also use `String.ascii_only?`. In either case, we need to read the file contents.
Other option I found is by using `Open3`. I tried them and chose the last one based on these Benchmark results.

```
def open3
  file_details, status = Open3.capture2e("file", cert_path)
  file_details.rpartition(":").last.strip == "PEM certificate"
end

def ascii_only
  File.read(cert_path).ascii_only?
end

Benchmark results:

open3 => 0.03903859999263659
ascii_only => 11.53894020000007
```

Reply:
`open3` didn't solved the purpose. We were still relying on shelling. powershell_out! is used since the command relies on `file.exe` which is available on powershell, but not on cmd.